lib/src/main/java/uk/gov/di/ipv/cri/address/library/exceptions/ValidationException.java

```diff
@@ -0,0 +1,11 @@
+package uk.gov.di.ipv.cri.address.library.exceptions;
+
+public class ValidationException extends Exception {
```

Comment:
Should some of these validation exceptions be specific to the class that is throwing them to make things easier to debug, or should we try to be as generic as possible?

Reply:
I agree maybe we could do something like this

```
public GenericException extends Exception {
      public GenericException(String errorMessage, Throwable cause) {
               super(errorMessage, cause);
      }
}
```

Then we can do this

```
new GenericException(ex.getMessage(), ex.getCause());
```

Reply:
```
public class PostcodeLookupProcessingException extends RuntimeException {
    public PostcodeLookupProcessingException() {
        super();
    }

    public PostcodeLookupProcessingException(String message) {
        super(message);
    }

    public PostcodeLookupProcessingException(String message, Throwable cause) {
        super(message, cause);
    }

    public PostcodeLookupProcessingException(Throwable cause) {
        super(cause);
    }
}
```

We use exceptions like this within the address API already

Reply:
I've moved these two exceptions to the `exception` package and renamed them to be more descriptive